plugins/kafka/src/api/KafkaDashboardClient.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { KafkaDashboardApi } from './types';
+import { Entity } from '@backstage/catalog-model';
+import { ConfigApi } from '@backstage/core-plugin-api';
+import { KAFKA_DASHBOARD_URL } from '../constants';
+
+export class KafkaDashboardClient implements KafkaDashboardApi {
+  private readonly configApi: ConfigApi;
+
+  constructor(options: { configApi: ConfigApi }) {
+    this.configApi = options.configApi;
+  }
+
+  getDashboardUrl(clusterId: string, entity: Entity): string {
+    const annotation = entity.metadata.annotations?.[KAFKA_DASHBOARD_URL] ?? '';
+
+    return (
+      (annotation &&
+        annotation
+          .split(',')
+          .map(value => {
+            const [clusterName, ...url] = value.split('/');
+            return { clusterName, url: url.join('/') };
+          })
+          .filter(({ clusterName }) => clusterId === clusterName)
+          .map(({ url }) => url)[0]) ||
```

Comment:
Is there a reason for allowing multiple urls in the annotation but only returning one from `getDashboardUrl`?
 Perhaps it's better to rename the annotation to `kafka.apache.org/dashboard-url` which also align better with the constant named `KAFKA_DASHBOARD_URL`

Reply:
The annotation `kafka.apache.org/dashboard-urls` can hold multiple dashboard urls, for each configured cluster one. I thought it's aligned with the annotation `kafka.apache.org/consumer-groups` as it holds multiple consumer groups as well. The `getDashboardUrl` can only return zero or one dashboard url as a cluster name should be unique.

Maybe this will be extended with the consumer group, see conversation with @marcus-crane.